object_pool.go

```diff
@@ -0,0 +1,45 @@
+package astits
+
+import "sync"
+
+// poolOfTempPayload global variable is used to ease access to pool from any place of the code
+var poolOfTempPayload = &poolTempPayload{
+	sp: sync.Pool{
+		New: func() interface{} {
+			// Prepare the slice of somewhat sensible initial size to minimize calls to runtime.growslice
+			return &tempPayload{
+				s: make([]byte, 0, 1024),
+			}
+		},
+	},
+}
+
+// tempPayload is an object containing payload slice
+type tempPayload struct {
```

Comment:
Why not use []byte directly instead of a struct containing []byte?

Reply:
Sorry for Japanese [resource](https://qiita.com/peroxyacyl/items/5e02ddf4480ecd2ec7b1) but code and benchmarks are still in English.
tl;dr We can pass slices directly, but it lead to additional allocations, since sync.pools are optimised to work with pointers. We can pass *[]byte object from/to pool, but we should track original pointer and update it accordingly. 

In my opinion, it's much simpler and safer to wrap []byte in some struct or use &bytes.Buffer{}.

Reply:
Additionally there is good Chinese [article](https://blog.csdn.net/FishGoddess/article/details/118254178) ([translation](https://programs.wiki/wiki/an-exploration-caused-by-a-pit-in-go-object-pool.html)) on this topic.

Reply:
OK in that case could you rename this struct to `bytesPoolItem`?

Reply:
Done